saleor/plugins/correios/plugin.py

```diff
@@ -0,0 +1,241 @@
+                        "Calculate Price and Delivery time URL need to be set",
+                        code=PluginErrorCode.INVALID.value,
+                    )
+                }
+            )
+        if "service_type" not in configuration or not configuration["service_type"]:
+            raise ValidationError(
+                {
+                    "service_type": ValidationError(
+                        "Correios Service Type is missing",
+                        code=PluginErrorCode.NOT_FOUND.value,
+                    )
+                }
+            )
+        if configuration["service_type"] not in CorreiosServiceType.list():
+            raise ValidationError(
+                {
+                    "service_type": ValidationError(
+                        f"Service Type must be one of {CorreiosServiceType.list()}",
+                        code=PluginErrorCode.INVALID.value,
+                    )
+                }
+            )
+
+    def _skip_plugin(self) -> bool:
+        """Rules to skip plugin."""
+        return not self.active
+
+    def _get_variant_attributes(
+        self, variant: "ProductVariant"
+    ) -> List[VariantAttribute]:
+        result: List[VariantAttribute] = list()
+        wanted_attributes = [
+            # (slug, unit)
+            ("weight", "kg"),
+            ("length", "cm"),
+            ("width", "cm"),
+            ("height", "cm"),
+        ]
+        for assigned_variant_attribute in variant.attributes.all():
+            attribute = assigned_variant_attribute.attribute
+            if (attribute.slug, attribute.unit) in wanted_attributes:
+                attribute_value = assigned_variant_attribute.values.first()
+                variant_attribute = VariantAttribute(
+                    slug=attribute.slug,
+                    unit=attribute.unit,
+                    value=attribute_value.name or attribute_value.value,
+                )
+                result.append(variant_attribute)
+        return result
+
+    def _group_checkout_lines_by_warehouse(
+        self, lines: List["CheckoutLineInfo"]
+    ) -> List[CheckoutLineWarehouse]:
+        warehouse_checkout_line_map: Dict[str, Any] = {}
+        for checkout_line_info in lines:
+            line = checkout_line_info.line
+            variant = line.variant
+            product = variant.product
+            stock = variant.stocks.get()
+            warehouse = stock.warehouse
+            warehouse_id = str(warehouse.id)
+            if warehouse_id not in warehouse_checkout_line_map:
+                warehouse = Warehouse(
+                    id=warehouse_id,
+                    name=warehouse.name,
+                    postal_code=warehouse.address.postal_code,
+                )
+                warehouse_checkout_line_map[warehouse_id] = CheckoutLineWarehouse(
+                    warehouse=warehouse, quantity=line.quantity, products=list()
```

Comment:
Is `quantity` supposed to be at this level? Shouldn't it be alongside (same cardinality as) each `product`?

Also, in order to keep names consistent:

```suggestion
                    vendor=vendor, lines=list()
```